src/index.ts

```diff
@@ -8,33 +8,168 @@ export interface SnowFlake {
+ * Sent when a message is create/updated. The inner payload is a message object.
  */
-export interface ReadMessage {
+export interface MessageEvent {
+    /**
+     * id of the message
+     */
+    id: SnowFlake;
 
     /**
-     * The message contents (up to 2000 characters)
+     * id of the channel the message was sent in
      */
-    content: string;
+    channel_id: SnowFlake;
+
+    /**
+     * id of the guild the message was sent in
+     */
+    guild_id?: SnowFlake;
+
+    /**
+     * user object
+     */
+    author: User;
 
     /**
-     * A nonce that can be used for optimistic message sending
+     * contents of the message
      */
-    nonce: SnowFlake;
+    content: string;
 
     /**
-     * true if this is a TTS message.
+     * whether this was a TTS message
      */
     tts: boolean;
 
     /**
-     * The contents of the file being sent
+     * any attached files
+     */
+    attachments: Attachment[];
+
+    /**
+     * used for validating a message was sent
+     */
+    nonce?: SnowFlake;
+
+    /**
+     * whether this message is pinned
+     */
+    pinned: boolean;
+
+    // TODO Add ?Guild Member
+
+    // TODO Timestamp
+
+    // TODO ?Editied_Timestamp
+
+    // TODO embed
+
+    // TODO mentions
+
+    // TODO mention_everyone
+
+    // TODO mention_roles
+
+    // TOOD ?reactions
+
+    // TOOD webhook_id
+
+    /**
+     * Type of the message
+     */
+    type: "create" | "edit";
+}
+
+/**
+ * Sent when a message is deleted.
+ * Sent when multiple messages are deleted at once.
+ */
+export interface MessageDeleteEvent {
+    /**
+     * the ids of the message
+     */
+    id: SnowFlake[];
+
+    /**
+     * When message delete bulk happens
+     */
+    isBulk: boolean;
+
+    /**
+     * the id of the channel
+     */
+    channel_id: SnowFlake;
+
+    /**
+     * the id of the guild
+     */
+    guild_id?: SnowFlake;
+}
+
+export interface User {
+    /**
+     * this user's id
+     */
+    id: SnowFlake;
+
+    /**
+     * the user's username, not unique across the platform
+     */
+    username: string;
+
+    /**
+     * the user's 4-digit discord-tag
+     */
+    discriminator: string;
+
+    /**
+     * the user's avatar hash
+     */
+    avatar: string;
+
+    /**
+     * the user's chosen language option
+     */
+    locale?: string;
+}
+
+/**
+ * The attachment interface
+ */
+export interface Attachment {
+    /**
+     * attachment id
+     */
+    id: SnowFlake;
+
+    /**
+     * name of file attached
+     */
+    filename: string;
+
+    /**
+     * size of file in bytes
+     */
+    size: number;
+
+    /**
+     * source url of file
+     */
+    url: string;
+
+    /**
+     * a proxied url of file
+     */
+    proxy_url: string;
+
+    /**
+     * height of file (if image)
      */
-    file: File;
+    height?: number;
 
     /**
-     * JSON encoded body of any additional request fields.
+     * width of file (if image)
      */
-    payload_json: string;
+    width: number;
```

Comment:
I read the Discord Docs API. And the width is not required.
It will be helpful if you made it to be like this
```ts
width: number | undefined;
```